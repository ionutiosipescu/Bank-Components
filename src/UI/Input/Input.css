/* ///////// Error Style Inputs ///////// */

input.input-error,
select.input-error {
  border-color: #fc8181;
}

/* ///////// Plan Page Card Style ///////// */

/* //////////////////////////////////  */
.box-exchange {
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid var(--green-light);
  transition: 0.3s;
  margin-top: 10px;
  margin-bottom: 10px;
  border-radius: 50px;
  box-shadow: 0px 0px 12px -8px rgba(21, 20, 47, 0.8);
  width: 150px;
  height: 40px;
}
.box-exchange:hover {
  scale: 1.08;
  background: var(--green);
  color: var(--white);
}
.box-exchange.selected {
  background: var(--green-dark);
  color: var(--white);
  border: 1px solid var(--green-dark);
}
.box-exchange.selected:hover {
  background: var(--green);
  color: var(--white);
  border: 1px solid var(--green);
}
/* //////////////////////////////////  */

.plan-form input[type="radio"] {
  display: none;
}
/* Plan Form Page Style */
.plan-form {
  display: flex;
  flex-direction: column;
  width: 100%;
}
/* //////////////////////////////////  */

/* Radio Selected Style */
.plan-form .radio-label.selected {
  border: 2px solid var(--blue);
  box-shadow: 0 15px 30px rgba(84, 197, 235, 0.6),
    0 10px 10px rgba(84, 197, 235, 0.2);
  background-color: var(--purple);
}
.plan-form .radio-label.selected .plancard-header h4,
.plan-form .radio-label.selected .plancard-header h1,
.plan-form .radio-label.selected .plancard-header .price-tag,
.plan-form .radio-label.selected .plancard-body li span {
  color: var(--white);
}
.plan-form .radio-label.selected .plancard-body li {
  color: var(--white);
  border-bottom: 1px solid var(--white);
}

.plan-form .radio-label.selected .plancard-footer {
  border: 1px solid var(--white);
  background: var(--white);
  color: var(--purple);
}
.plan-form .radio-label.selected .plancard-footer:hover {
  background: var(--purple);
  color: var(--white);
}

/* Card Style */
.plan-form .radio-label {
  position: relative;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  display: flex;
  width: 260px;
  height: auto;
  margin: 10px;
  padding: 20px;
  border-radius: 4px;
  font-size: 14px;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 25px;
  overflow: hidden;
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23);
}

/* Header */
.plan-form .plancard-header {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.plan-form .plancard-header > h4 {
  font-size: 1rem;
  color: var(--purple);
  margin-top: 20px;
  margin-bottom: 0px;
}

.plan-form .plancard-header .plancard-price {
  margin-right: 5px;
  font-size: 15px;
  font-weight: 300;
  display: flex;
  justify-content: center;
  align-items: center;
}
.plan-form .plancard-header > h1 {
  margin: 0;
}
.plan-form .plancard-header .price-tag {
  margin: 0 2px 16px 0;
}
.plan-form .plancard-price h1:after {
  content: "/mo";
  margin-left: 5px;
  font-size: 15px;
  font-weight: 300;
}

/* Body */
.plan-form .plancard-body {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  z-index: 2;
  margin-bottom: 20px;
}
.plan-form .plancard-body ul {
  list-style-type: none;
  padding: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  gap: 20px;
  text-align: center;
}
.plan-form .plancard-body li {
  width: 100%;
  color: var(--gray-dark);
  border-bottom: 1px solid var(--gray-dark);
  padding-bottom: 8px;
}
.plan-form .plancard-body li:last-child {
  border: none;
}
.plan-form .plancard-body li span {
  margin-left: 10px;
  color: black;
}

/* Footer */
.plan-form .plancard-footer {
  height: 30px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 50px;
  border: 1px solid var(--blue);
  background: var(--purple);
  color: var(--white);
  transition: 0.3s;
  margin: 10px 0 10px 0;
  cursor: pointer;
  padding: 18px;
}
.plan-form .plancard-footer:hover {
  background: var(--white);
  color: var(--blue);
}
