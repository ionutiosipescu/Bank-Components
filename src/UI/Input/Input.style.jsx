import styled from "styled-components";
import DatePicker from "react-datepicker";
import { device } from "../../../utils/breakpoints/breakpoints";
import { css } from "styled-components";
import ReCAPTCHA from "react-google-recaptcha";

const StyledGroup = css`
  position: relative;
  display: flex;
  flex-direction: column;
  margin: 0 10px 10px 0;
  width: ${(props) => (props.small ? "200px" : "100%")};
  /* label {
    ${(props) =>
    props.large ? " margin-bottom: 20px;" : " margin-bottom: 10px;"};
  } */
  @media (${device.mobileS}) {
    label {
      ${(props) =>
        props.large ? " margin-bottom: 10px;" : " margin-bottom: 10px;"};
    }
  }
  @media (${device.laptop}) {
    label {
      ${(props) =>
        props.large ? " margin-bottom: 20px;" : " margin-bottom: 10px;"};
    }
  }
`;

export const Group = styled.div`
  ${StyledGroup}
  color: var(--text);
  label:after {
    content: "*";
    color: red;
  }
`;
export const ReCaptchaGroup = styled.div`
  ${StyledGroup}
  margin-top: 20px;
`;

export const StyledReCAPTCHA = styled(ReCAPTCHA)``;

export const GroupRadio = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  margin: 0 10px 10px 0;
  width: 200px;
  label {
    margin-bottom: 10px;
    margin-right: 10px;
  }
  @media (max-width: 1680px) {
    ${(props) =>
      props.small
        ? "width: 150px; height: 40px;"
        : props.large
        ? "width: 300px; height: 40px;"
        : props.radio
        ? "width: 200px; height: auto"
        : "width: 300px; height: 40px;"}
  }
`;
export const GroupRadioRegister = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  ${(props) => console.log(props)}
  margin: 0 10px 10px 0;
  width: 100%;
  label {
    /* margin-bottom: 10px;
    margin-right: 10px; */
    width: 100%;
  }
  @media (max-width: 1680px) {
    ${(props) =>
      props.small
        ? "width: 150px; height: 40px;"
        : props.large
        ? "width: 300px; height: 40px;"
        : props.radio
        ? "width: 200px; height: auto"
        : "width: 300px; height: 40px;"}
  }
`;

export const GroupRadioOption = styled.div`
  display: flex;
  width: 200px;
  justify-content: flex-start;
  align-items: center;
  input {
    margin-right: 15px;
  }
`;
export const RadioLabel = styled.label`
  margin-bottom: 25px;
  :after {
    content: "*";
    color: red;
  }
`;
export const RadioPlanContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: ${(props) => (props.size === "card" ? "0" : "20px")}; // 50px
  justify-content: ${(props) =>
    props.size === "card" ? "space-between" : "center"};
  align-items: center;
  @media ${device.laptop} {
    flex-direction: row;
    gap: 20px;
  }
`;

// shared style
const InputStyle = css`
  height: ${(props) => (props.tall ? "60px" : "35px")};
  border: 1px solid var(--purple);
  border-radius: 30px;
  padding: 0 12px 0 10px;
  background-color: var(--white);
  &:focus {
    outline: none;
    border: 2px solid var(--blue);
  }
`;

// lv1
export const InputForm = styled.input`
  ${InputStyle}
  width: ${(props) => (props.small ? "200px" : "100%")};

  ${(props) =>
    props.large
      ? "height: 60px; padding-left: 20px; font-size: 1.1rem; width: 100%;"
      : "padding-left: 20px;"}
  @media (${device.mobileS}) {
    ${(props) =>
      props.large
        ? "height: 40px; padding-left: 20px; font-size: 1.1rem; width: 100%;"
        : ""}
  }
  @media (${device.mobileL}) {
    ${(props) =>
      props.large
        ? "height: 50px; padding-left: 20px; font-size: 1.1rem; width: 100%;"
        : ""}
  }
  @media (${device.laptop}) {
    ${(props) =>
      props.large
        ? "height: 60px; padding-left: 20px; font-size: 1.1rem; width: 100%;"
        : ""}
  }
`;

export const InputSelect = styled.select`
  ${InputStyle}
  width: ${(props) => (props.small ? "200px" : "100%")};
`;

export const InputRadio = styled.input``;

export const Checkbox = styled.div`
  position: relative;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin: 15px 0px 15px;
  input {
    width: fit-content;
    margin: 0 10px 0 0;
    transform: scale(1.25);
  }
`;

export const Error = styled.div`
  color: #fc8181;
  font-size: 0.75rem;
  margin-top: 0.25rem;
  position: absolute;
  bottom: -22px;
`;
