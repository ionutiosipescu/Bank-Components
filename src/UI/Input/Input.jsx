import React from "react";
import { InputForm, Group } from "./Input.style";

function Input({ label, min, max, ...otherProps }) {
  return (
    <Group {...otherProps}>
      <label>{label}</label>
      <InputForm {...otherProps} min={min} max={max} />
    </Group>
  );
}
export default Input;
