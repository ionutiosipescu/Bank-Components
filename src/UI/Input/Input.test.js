import React from "react";
import { shallow } from "enzyme";
import Input from "./Input";
import { InputForm, Group } from "./Input.style";

describe("Input component", () => {
  it("should render with correct props", () => {
    const props = {
      label: "Name",
      type: "text",
      value: "John",
      onChange: jest.fn(),
      required: true,
      min: 0,
      max: 100,
    };

    const wrapper = shallow(<Input {...props} />);

    expect(wrapper.find(Group)).toHaveLength(1);
    expect(wrapper.find(InputForm)).toHaveLength(1);
    expect(wrapper.find("label").text()).toEqual("Name");
    expect(wrapper.find(InputForm).props().type).toEqual("text");
    expect(wrapper.find(InputForm).props().value).toEqual("John");
    expect(wrapper.find(InputForm).props().onChange).toEqual(props.onChange);
    expect(wrapper.find(InputForm).props().required).toEqual(true);
    expect(wrapper.find(InputForm).props().min).toEqual(0);
    expect(wrapper.find(InputForm).props().max).toEqual(100);
    expect(wrapper).toMatchSnapshot();
  });
});
