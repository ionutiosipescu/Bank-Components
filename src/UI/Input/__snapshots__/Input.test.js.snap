// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render with correct props 1`] = `
<styled.div
  onChange={[MockFunction]}
  required={true}
  type="text"
  value="John"
>
  <label>
    Name
  </label>
  <styled.input
    max={100}
    min={0}
    onChange={[MockFunction]}
    required={true}
    type="text"
    value="John"
  />
</styled.div>
`;
